opentelemetry-etw-metrics/src/exporter/mod.rs

```diff
@@ -39,265 +41,146 @@ impl Debug for MetricsExporter {
     }
 }
 
+fn emit_export_metric_service_request(
+    export_metric_service_request: &ExportMetricsServiceRequest,
+    encoding_buffer: &mut Vec<u8>,
+) -> MetricResult<()> {
+    if (export_metric_service_request.encoded_len()) > etw::MAX_EVENT_SIZE {
+        otel_warn!(name: "MetricExportFailedDueToMaxSizeLimit", size = export_metric_service_request.encoded_len(), max_size = etw::MAX_EVENT_SIZE);
+    } else {
+        encoding_buffer.resize_with(
+            export_metric_service_request.encoded_len(),
+            Default::default,
+        );
+
+        export_metric_service_request
+            .encode(encoding_buffer)
+            .map_err(|err| MetricError::Other(err.to_string()))?;
+
+        let result = etw::write(encoding_buffer);
+        // TODO: Better logging/internal metrics needed here for non-failure
+        // case Uncomment the line below to see the exported bytes until a
+        // better logging solution is implemented
+        // println!("Exported {} bytes to ETW", byte_array.len());
+        if result != 0 {
+            otel_warn!(name: "MetricExportFailed", error_code = result);
+        }
+    }
+
+    Ok(())
+}
+
 #[async_trait]
 impl PushMetricExporter for MetricsExporter {
     async fn export(&self, metrics: &mut ResourceMetrics) -> MetricResult<()> {
+        let schema_url: String = metrics
+            .resource
+            .schema_url()
+            .map(Into::into)
+            .unwrap_or_default();
+
+        let mut encoding_buffer = Vec::new();
```

Comment:
@mattbodd Please add a TODO here to avoid this vec allocation, and instead use a mutex wrapped vec, pre-created in this struct.